receiver/elasticsearchreceiver/scraper_test.go

```diff
@@ -0,0 +1,346 @@
+	require.NoError(t, json.Unmarshal(healthJSON, &clusterHealth))
+
+	return &clusterHealth
+}
+
+func nodeStats(t *testing.T) *model.NodeStats {
+	nodeJSON, err := ioutil.ReadFile("./testdata/sample_payloads/nodes_linux.json")
+	require.NoError(t, err)
+
+	nodeStats := model.NodeStats{}
+	require.NoError(t, json.Unmarshal(nodeJSON, &nodeStats))
+	return &nodeStats
+}
+
+// TestWriteGolden writes golden metrics to testdata; It should only be used to generate golden metrics when changes to metrics are made.
+func TestWriteGolden(t *testing.T) {
+	t.SkipNow()
+
+	sc := newElasticSearchScraper(zap.NewNop(), createDefaultConfig().(*Config))
+
+	err := sc.start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+
+	mockClient := mocks.MockElasticsearchClient{}
+	mockClient.On("ClusterHealth", mock.Anything).Return(clusterHealth(t), nil)
+	mockClient.On("NodeStats", mock.Anything, mock.Anything).Return(nodeStats(t), nil)
+
+	sc.client = &mockClient
+
+	fullMetrics, err := sc.scrape(context.Background())
+	require.NoError(t, err)
+
+	sc.cfg.SkipClusterMetrics = true
+	clusterSkipMetrics, err := sc.scrape(context.Background())
+	require.NoError(t, err)
+	sc.cfg.SkipClusterMetrics = false
+
+	sc.cfg.Nodes = []string{}
+	noNodesMetrics, err := sc.scrape(context.Background())
+	require.NoError(t, err)
+	sc.cfg.Nodes = []string{"_all"}
+
+	initFileDir := func(t *testing.T, filePath string) {
+		dir := filepath.Dir(filePath)
+		mkdirErr := os.MkdirAll(dir, 0777)
+		require.NoError(t, mkdirErr)
+	}
+
+	initFileDir(t, fullExpectedMetricsPath)
+	initFileDir(t, skipClusterExpectedMetricsPath)
+	initFileDir(t, noNodesExpectedMetricsPath)
+
+	err = golden.WriteMetrics(fullExpectedMetricsPath, fullMetrics)
+	require.NoError(t, err)
+
+	err = golden.WriteMetrics(skipClusterExpectedMetricsPath, clusterSkipMetrics)
+	require.NoError(t, err)
+
+	err = golden.WriteMetrics(noNodesExpectedMetricsPath, noNodesMetrics)
+	require.NoError(t, err)
+}
+
+func requireMetricsEqual(t *testing.T, m1, m2 pdata.Metrics) {
+	rms1 := m1.ResourceMetrics()
+	rms2 := m2.ResourceMetrics()
+	if rms1.Len() != rms2.Len() {
+		require.Fail(t, "First metric had %d resource metrics, second had %d", rms1.Len(), rms2.Len())
+	}
+
+	for i := 0; i < rms1.Len(); i++ {
+		rm1 := rms1.At(i)
+		rm2 := rms2.At(i)
+
+		require.Equal(t, rm1.Resource().Attributes().AsRaw(), rm2.Resource().Attributes().AsRaw())
+
+		ilms1 := rm1.InstrumentationLibraryMetrics()
+		ilms2 := rm2.InstrumentationLibraryMetrics()
+
+		if ilms1.Len() != ilms2.Len() {
+			require.Fail(t, "Resource metric %d: First metric had %d ILL metrics, second had %d", i, ilms1.Len(), ilms2.Len())
+		}
+
+		for j := 0; j < ilms1.Len(); j++ {
+			ilm1 := ilms1.At(j)
+			ilm2 := ilms2.At(j)
+
+			require.Equal(t, ilm1.InstrumentationLibrary().Name(), ilm2.InstrumentationLibrary().Name())
+
+			err := scrapertest.CompareMetricSlices(ilm1.Metrics(), ilm2.Metrics())
+			require.NoError(t, err)
+		}
+	}
+}
```

Comment:
This would be a great contribution to `scrapertest`. We need to have full comparison at `pdata.Metrics` level eventually.

Reply:
Agreed. It really only becomes urgent when we are about to have multiple implementations. If you'll create a tracking issue and link it here, I'll resolve this thread.

Reply:
I think this issue made a while back is sufficient: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/6890